util/webhook/validator_test.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateBoolPtrImmutable(t *testing.T) {
+	g := NewWithT(t)
+	testPath := field.NewPath("Spec", "Foo")
+
+	tests := []struct {
+		name           string
+		path           *field.Path
+		input1         *bool
+		input2         *bool
+		expectedOutput *field.Error
+	}{
+		{
+			name:   "nil",
+			path:   testPath,
+			input1: nil,
+			input2: nil,
+		},
+		{
+			name:   "no change",
+			path:   testPath,
+			input1: nil,
+			input2: nil,
+		},
```

Comment:
Since this test case is the same as the one above it, I wonder if setting the inputs to the same non-nil value might provide a bit more value.

Reply:
💯